include/boost/ut.hpp

```diff
@@ -1289,6 +1293,10 @@ class runner {
         test();
 #if defined(__cpp_exceptions)
       } catch (const events::fatal_assertion&) {
+      } catch (const std::exception& e) {
+        std::runtime_error errorMsg(e.what());
```

Comment:
I'm unsure about this conversion of every std::exception type to std::runtime_error...

This is probably because I'm used to test frameworks saying things like:

"Unexpected exception of type MyCustomExceptionType occurred, with 'My informative explanation of the problem'"

The type of the exception can be really useful in narrowing down the cause, or location.

If it isn't possible to preserve the exception type here, I wonder whether saving the type-name of the exception in a std::string would be possible instead? It would likely be less confusing and more useful.

Reply:
I used std::runtime_error because it has a constructor that takes a message and inherits from std::exception (which is basically what ApprovalException does for example) and it doesn't add a new dependency, and it also fulfils the initial goal of printing the exception message. To keep the original type of the exception would be trickier, and I haven't figured out a good way for that. If you use the throw<ExceptionType> function you can check for the type of the exception, but it's not the use case I had in mind (just printing the ApprovalTests message).